R/library_call_linter.R

```diff
@@ -94,13 +114,13 @@ library_call_linter <- function(allow_preamble = TRUE) {
         and not(ancestor::expr[FUNCTION])
       )
     ]
-  "
+  ")
 
   bad_indirect_funs <- c("do.call", "lapply", "sapply", "map", "walk")
-  call_symbol_cond <- "
-  SYMBOL[text() = 'library' or text() = 'require']
-    or STR_CONST[text() = '\"library\"' or text() = '\"require\"']
-  "
+  call_symbol_cond <- glue("
+    SYMBOL[{attach_call_cond}]
+    or STR_CONST[{ xp_text_in_table(dQuote(attach_calls, '\"')) }]
```

Comment:
This is to catch `lapply(pkgs, "library", character.only = TRUE)`?
Is that worth it? If yes, we should catch `lapply(pkgs, 'library', character.only = TRUE)` too.

Reply:
Yes, but let's mark as follow-up since it's not related to consecutive suppression but `character.only`

Reply:
#2319